src/main/java/com/agency04/sbss/pizza/service/impl/PizzaDeliveryServiceConstructorInjection.java

```diff
@@ -0,0 +1,45 @@
+package com.agency04.sbss.pizza.service.impl;
+
+import com.agency04.sbss.pizza.model.Pizza;
+import com.agency04.sbss.pizza.model.PizzaIngredient;
+import com.agency04.sbss.pizza.service.PizzaDeliveryService;
+import com.agency04.sbss.pizza.service.PizzeriaService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import java.util.stream.Collectors;
+
+/**
+ * Concrete pizza delivery service
+ */
+@Component
+public class PizzaDeliveryServiceConstructorInjection implements PizzaDeliveryService {
+
+	/**
+	 * Pizzeria service that handles pizza orders
+	 */
+	private PizzeriaService pizzeriaService;
+
+	@Autowired
+	public PizzaDeliveryServiceConstructorInjection(@Qualifier("firstPizzeria") PizzeriaService pizzeriaService) {
+		this.pizzeriaService = pizzeriaService;
+	}
+
+	@Override
+	public String orderPizza(Pizza pizza) {
+		String order = "";
+
+		order += pizzeriaService.getName() + " Pizzeria: "
```

Comment:
Nije velika greška, ali može biti one-liner, ne dobivaš ništa posebnom definicijom i naknadnim appendanjem stringova. `String order = pizzeriaService.getName() + " Pizzeria: " ...` , nešto čišći kod

Reply:
U nekim slučajevima bi ti čak IDE javio automatski da možeš jednostavno iza return konkatenirati te sve vrijednosti